### DISCUSSION TITLE
Local app creation in Saleor

### DISCUSSION THREAD

**Discussion by satvikt** [saleor/saleor#9132] [General]
I wish to write a local app - say "sample_app" - for my Saleor use case, but I am unable to determine the location and initiation point. I must say, the current docs for Saleor 3.0 lacks a thorough read for extending it via apps..

Here are my questions, to start with

1. Where will the app folder/package be located, to write the app code?
2. I was able to create an app using  "create_app" command, but I am unable to find the package/folder associated with it anywhere in my saleor-core installation
3. what is the package structure of an app?
4. Apart from the manifest, what are the other components required to create an app?
5. the sample manifest at https://docs.saleor.io/docs/3.0/developer/extending/apps/manifest, shows the following URL:
`  "appUrl": "http://localhost:3000/app",`
However, there's nothing running at port 3000.

**Comment by korycins** [comment]
Saleor App is a separate instance(service) that communicates with Saleor via API and receives updates from Saleor via webhooks.
The app can also extend a dashboard by mounting its own iframe views. 
App docs: https://docs.saleor.io/docs/3.0/developer/extending/apps/key-concepts
Our framework for building apps. You can find there app samples: https://github.com/saleor/saleor-app-framework-python

**Reply by MacLake** [reply]
I’m also struggling a bit with the term “create a local app”.  If you “create a product” in or any Saleor any shop software, it’s clear that you just create an instance of a model in the database and not an actual product. But for an app the term might be misleading.

So did I get it right when I assume that “creating an app” just means that an instance of a model is being created where you can define tokens and webhooks is being created and that the real app lives in another project (e. g. another Django project?

**Reply by korycins** [reply]
> So did I get it right when I assume that “creating an app” just means that an instance of a model is being created where you can define tokens and webhooks is being created and that the real app lives in another project (e. g. another Django project?

Yes, exacly

**Reply by MacLake** [reply]
Thanks for clarifying this!
